packages/sdk/src/assets/bridge-asset.ts

```diff
@@ -0,0 +1,267 @@
+
+    return {
+      destinationGasPrice,
+      destinationGasLimit,
+    }
+  }
+
+  async estimateBridgeOut(
+    chainId: number,
+    toChainId: number,
+    contractAddress: Address,
+    bridgeOutPayload: Hex,
+  ): Promise<EstimateBridgeOutResult> {
+    const {destinationGasPrice, destinationGasLimit} = await this._estimateBridgeOutDestinationGas(
+      chainId,
+      toChainId,
+      contractAddress,
+      bridgeOutPayload,
+    )
+
+    const holographToChainId = getNetworkByChainId(toChainId).holographId
+
+    // now that we have gasLimit and gasPrice, we update the payload to include the proper data needed for estimating fees
+    const bridgeRequestPayload = (
+      await this._createBridgeOutPayload(
+        chainId,
+        holographToChainId,
+        contractAddress,
+        destinationGasLimit,
+        destinationGasPrice,
+        bridgeOutPayload,
+      )
+    ).result
+
+    // we are now ready to get fees for transaction
+    const feeValue = await this._getValue(
+      chainId,
+      holographToChainId,
+      destinationGasLimit,
+      destinationGasPrice,
+      bridgeRequestPayload,
+    )
+
+    const unsignedBridgeOutTx = BridgeAsset.createUnsignedBridgeOutTx(
+      holographToChainId,
+      contractAddress,
+      destinationGasLimit,
+      destinationGasPrice,
+      bridgeOutPayload,
+      //{value: feeValue}, TODO: do we need to pass the value here?
+    )
+
+    // Source transaction parameters
+
+    const sourceGasPrice = this.gasSettings?.sourceGasPrice ?? (await this._providers.byChainId(chainId).getGasPrice())
+
+    const sourceGasLimit = this.gasSettings?.sourceGasLimit ?? BigInt(450000) // NOTE: Only to test user balance
+    /**
+     * TODO: put this in another place, this does not make part of the function scope
+      const userBalance = await this.getSignerBalance()
+      const sourceTxGas = sourceGasPrice.mul(sourceGasLimit)
+      const totalValueNeeded = sourceTxGas.add(feeValue)
+      const error = checkBalanceBeforeTX(userBalance, totalValueNeeded) // TODO: throw error for balance check
+     */
+
+    return {
+      gasSource: {
+        chainId,
+        gasPrice: sourceGasPrice,
+        gasLimit: sourceGasLimit,
+      },
+      gasDestination: {
+        chainId: toChainId,
+        gasPrice: destinationGasPrice,
+        gasLimit: destinationGasLimit,
+      },
+      value: feeValue,
+      unsignedTx: unsignedBridgeOutTx,
+    }
+  }
+
+  protected async _bridgeOut(
+    chainId: number,
+    toChainId: number,
+    wallet: HolographWallet,
+    contractAddress: Address,
+    bridgeOutPayload: Hex,
+    options?: {sourceGasPrice?: bigint; sourceGasLimit?: bigint},
+  ): Promise<TransactionReceipt> {
+    const {unsignedTx, gasSource, value} = await this.estimateBridgeOut(
+      chainId,
+      toChainId,
+      contractAddress,
+      bridgeOutPayload,
+    )
+
+    const walletClient = wallet.onChain(chainId)
+
+    const tx = await walletClient.sendTransaction({
```

Comment:
This approach is different from what we've been doing lately.

We've been using the writeContract method from HolographBase instead of  `walletClient.sendTransaction`. Let's choose only one option for consistency.

We can also pass overrides to the writeContract methods, such as gas, gasPrice, value etc. You can take a look at how we invoked functions such as `mint`, `deploy` etc

Reply:
Ah ok, because of the `data` argument right, I think using writeContract we don't support this argument. So it's all good. 

Then let's use this approach only in this case, and keep using the writeContract as the "default" approach